packages/graphql_server/src/dataAggregation.ts

```diff
@@ -0,0 +1,126 @@
+import { GitHubOrganization, GitHubInfo } from '../types/githubApi.types'
+import { getOrganizationInfo, getRepoInfo } from './api/githubApi'
+import { OrganizationInsertion, ProjectInsertion } from '../types/dataAggregation.types'
+import supabase from './supabase'
+
+/**
+ * Returns the info for a repository in a format that can be inserted into the DB.
+ * It goes through all data sources and aggregates the data.
+ * @param {string} name - The name of the repository.
+ * @param {string} owner - The name of the organization.
+ * @returns {ProjectInsertion | null} The info for the repository in a format that can be inserted into the DB.
+ */
+export const aggregateDataForRepo = async (name: string, owner: string) => {
+  let repoInfo: ProjectInsertion | null = null
+  // query send to github. If this is changed the corresponding types have to be changed as well
+  const query = `query {
+    repository(owner: "${owner}", name: "${name}") {
+      name 
+      description
+      stargazerCount
+      issues(filterBy: {states: [OPEN]}) {totalCount}
+      forkCount
+      pullRequests(states: [OPEN]) {totalCount}
+      url
+      homepageUrl
+    owner {
+      login
+    }
+  }
+  }`
+
+  // call github api
+  const repoGHdata: GitHubInfo | null = await getRepoInfo(
+    query,
+    'Bearer ' + process.env.GITHUB_API_TOKEN
+  )
+
+  if (repoGHdata === null) {
+    console.error('Could not get GitHub data for repo', name)
+    return null
+  }
+
+  const organizationID = await getOrganizationID(owner)
+
+  // @Todo aggregate more data for the repo
+
+  repoInfo = {
+    name: repoGHdata.name,
+    about: repoGHdata.description,
+    star_count: repoGHdata.stargazerCount,
+    issue_count: repoGHdata.issues.totalCount,
+    fork_count: repoGHdata.forkCount,
+    pull_request_count: repoGHdata.pullRequests.totalCount,
+    contributor_count: 1,
+    github_url: repoGHdata.url,
+    website_url: repoGHdata.homepageUrl,
+    owned_by: organizationID ?? '634b6eb5-30c8-4818-81d3-e1d98cb0b2c7',
```

Comment:
no

Reply:
this is just a übergangslösung. Right now when no organization can be found as an owner for example because the owner is a user, this id which belongs to vercel in the db is assigned as owner. 
I will change this today but would prefer to do so on a new branch where i also add other features. 
This PR should be merged asap